src/main/java/org/openstreetmap/atlas/checks/validation/tag/SourceMaxspeedCheck.java

```diff
@@ -0,0 +1,161 @@
+    private static final int COUNTRY_INSTRUCTION_INDEX = 2;
+    private static final int CONTEXT_INSTRUCTION_INDEX = 3;
+
+    private final List<String> exceptedCountries;
+
+    /**
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public SourceMaxspeedCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.exceptedCountries = this.configurationValue(configuration, "excepted_countries",
+                DEFAULT_EXCEPTIONS);
+    }
+
+    /**
+     * Valid objects for this check are Points and Edges with a source:maxspeed tag and are not part
+     * of the excepted countries.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return !this.isFlagged(object.getOsmIdentifier())
+                && (object instanceof Edge || object instanceof Point)
+                && this.hasSourceMaxspeed(object)
+                && (object.getTags().containsKey(ISOCountryTag.KEY) && !this.exceptedCountries
+                        .contains(object.tag(ISOCountryTag.KEY).toUpperCase()));
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Optional<String> sourceMaxspeed = object.getTag(SOURCE_MAXSPEED);
+        if (sourceMaxspeed.isPresent())
+        {
+            final Set<String> instructions = new HashSet<>();
+            final String sourceValue = sourceMaxspeed.get();
+            final Matcher matcher = COUNTRY_CONTEXT_PATTERN.matcher(sourceValue);
+            if (matcher.find())
+            {
+                final String[] parts = sourceValue.split(COLON);
+                if (!this.isCountryValid(parts[0]))
+                {
+                    instructions
+                            .add(this.getLocalizedInstruction(COUNTRY_INSTRUCTION_INDEX, parts[0]));
+                }
+                if (!this.isContextValid(parts[1]))
+                {
+                    instructions
+                            .add(this.getLocalizedInstruction(CONTEXT_INSTRUCTION_INDEX, parts[1]));
+                }
+
+            }
+
+            else if (!POSSIBLE_VALUES.contains(sourceValue) && !sourceValue.contains(ZONE))
+            {
+                instructions.add(this.getLocalizedInstruction(VALUE_INSTRUCTION_INDEX));
+            }
+            if (!instructions.isEmpty())
+            {
+                final CheckFlag flag = this.createFlag(object, this.getLocalizedInstruction(
```

Comment:
It would be great if edges that share an osm id could be grouped into one flag like [here](https://github.com/osmlab/atlas-checks/blob/dev/src/main/java/org/openstreetmap/atlas/checks/validation/tag/InvalidLanesTagCheck.java#L89). For this to work properly it is also important to mark items as flagged by there osm ids.

Reply:
Also what are your thoughts on embedding a fix suggestion here to convert the tag to `sign` or `markings`? @mm-ciub

Reply:
@Bentleysb I will start making the improvements, thank you.
@seancoulter, looking over taginfo I see that most cases have the value 'sign' at 21.30% while 'markings' is at 0.01%.  So perhaps a fix suggestion for sign could be added, but the user should assure that there is a sign present. Otherwise it falls under the zone source requiring the country:context format.

Reply:
Ok thanks for the explanation. sounds like this might involve more research, and then we can open up an Issue if fix suggestions are found to be suitable